test/unit/file_transfer_agent/azure_test.js

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+const assert = require('assert');
+const mock = require('mock-require');
+const SnowflakeAzureUtil = require('./../../../lib/file_transfer_agent/azure_util');
+const resultStatus = require('./../../../lib/file_transfer_agent/file_util').resultStatus;
+
+describe('AZ client', function () {
+  const mockDataFile = 'mockDataFile';
+  const mockLocation = 'mockLocation';
+  const mockTable = 'mockTable';
+  const mockPath = 'mockPath';
+  const mockDigest = 'mockDigest';
+  const mockKey = 'mockKey';
+  const mockIv = 'mockIv';
+  const mockMatDesc = 'mockMatDesc';
+
+  let AZ = null;
+  let azure = null;
+  let filestream = null;
+  const dataFile = mockDataFile;
+  const meta = {
+    stageInfo: {
+      location: mockLocation,
+      path: mockTable + '/' + mockPath + '/',
+      creds: {}
+    },
+    SHA256_DIGEST: mockDigest,
+  };
+  const encryptionMetadata = {
+    key: mockKey,
+    iv: mockIv,
+    matDesc: mockMatDesc
+  };
+
+  before(function () {
+    mock('azure', {
+      BlobServiceClient: function() {
+        function BlobServiceClient() {
+          this.getContainerClient = function () {
+            function getContainerClient() {
+              this.getBlobClient = function () {
+                function getBlobClient() {
+                  this.getProperties = function () {
+                    function getProperties() { 
+                      this.then = function (callback) {
+                        callback({
+                          metadata: {}
+                        });
+                      };
+                    }
+                    return new getProperties;
+                  };
+                }
+                return new getBlobClient;
+              };
+            }
+            return new getContainerClient;
+          };
+        }
+        return new BlobServiceClient;
+      }
+    });
+
+    mock('filestream', {
+      readFileSync: async function (data) {
+        return data;
+      }
+    });
+    
+    azure = require('azure');
+    filestream = require('filestream');
+    AZ = new SnowflakeAzureUtil(azure, filestream);
+  });
+
+  it('extract bucket name and path', async function () {
+    let result = AZ.extractContainerNameAndPath('sfc-dev1-regression/test_sub_dir/');
+    assert.strictEqual(result.containerName, 'sfc-dev1-regression');
```

Comment:
this test seems to be combination of some tests that can be parameterized

Reply:
updated.